tests/bot/utils/test_redis_cache.py

```diff
@@ -42,95 +50,105 @@ async def test_set_get_item(self):
-    #     self.redis.update({
-    #         "reckfried": "jona",
-    #         "mega": "hungry, though",
-    #     })
-    #
-    #     result = {
-    #         "reckfried": "jona",
-    #         "bel air": "prince",
-    #         "mega": "hungry, though",
-    #     }
-    #     self.assertEqual(self.redis.copy(), result)
+    async def test_set_item_type(self):
+        """Test that .set rejects keys and values that are not strings, ints or floats."""
+        fruits = ["lemon", "melon", "apple"]
+
+        with self.assertRaises(TypeError):
+            await self.redis.set(fruits, "nice")
+
+    async def test_delete_item(self):
+        """Test that .delete allows us to delete stuff from the RedisCache."""
+        # Add an item and verify that it gets added
+        await self.redis.set("internet", "firetruck")
+        self.assertEqual(await self.redis.get("internet"), "firetruck")
+
+        # Delete that item and verify that it gets deleted
+        await self.redis.delete("internet")
+        self.assertIs(await self.redis.get("internet"), None)
+
+    async def test_contains(self):
+        """Test that we can check membership with .contains."""
+        await self.redis.set('favorite_country', "Burkina Faso")
+
+        self.assertIs(await self.redis.contains('favorite_country'), True)
+        self.assertIs(await self.redis.contains('favorite_dentist'), False)
+
+    async def test_items(self):
+        """Test that the RedisDict can be iterated."""
+        await self.redis.clear()
+
+        # Set up our test cases in the Redis cache
+        test_cases = [
+            ('favorite_turtle', 'Donatello'),
+            ('second_favorite_turtle', 'Leonardo'),
+            ('third_favorite_turtle', 'Raphael'),
+        ]
+        for key, value in test_cases:
+            await self.redis.set(key, value)
+
+        # Consume the AsyncIterator into a regular list, easier to compare that way.
+        redis_items = [item async for item in self.redis.items()]
+
+        # These sequences are probably in the same order now, but probably
+        # isn't good enough for tests. Let's not rely on .hgetall always
+        # returning things in sequence, and just sort both lists to be safe.
+        redis_items = sorted(redis_items)
+        test_cases = sorted(test_cases)
+
+        # If these are equal now, everything works fine.
+        self.assertSequenceEqual(test_cases, redis_items)
+
+    async def test_length(self):
+        """Test that we can get the correct .length from the RedisDict."""
+        await self.redis.clear()
+        await self.redis.set('one', 1)
+        await self.redis.set('two', 2)
+        await self.redis.set('three', 3)
+        self.assertEqual(await self.redis.length(), 3)
+
+        await self.redis.set('four', 4)
+        self.assertEqual(await self.redis.length(), 4)
+
+    async def test_to_dict(self):
+        """Test that the .copy method returns a workable dictionary copy."""
```

Comment:
This docstring mentions the `.copy` method, but it's testing the `.to_dict` method.

Reply:
Oops. I'll fix that. :eye: